log4j-core/src/main/java/org/apache/logging/log4j/core/pattern/EncodingPatternConverter.java

```diff
@@ -86,100 +87,4 @@ public void format(final LogEvent event, final StringBuilder toAppendTo) {
-                            break;
-                        case '"':
-                            toAppendTo.setCharAt(i, '&');
-                            toAppendTo.insert(i + 1, "quot;");
-                            break;
-                        case '\'':
-                            toAppendTo.setCharAt(i, '&');
-                            toAppendTo.insert(i + 1, "apos;");
-                            break;
-                        case '/':
-                            toAppendTo.setCharAt(i, '&');
-                            toAppendTo.insert(i + 1, "#x2F;");
-                            break;
-                    }
-                }
-            }
-        },
-
-        /**
-         * JSON string escaping as defined in RFC 4627.
-         *
-         * @see <a href="https://www.ietf.org/rfc/rfc4627.txt">RFC 4627</a>
-         */
-        JSON {
-            @Override
-            void escape(final StringBuilder toAppendTo, final int start) {
-                for (int i = toAppendTo.length() - 1; i >= start; i--) { // backwards: length may change
-                    final char c = toAppendTo.charAt(i);
-                    if (Character.isISOControl(c)) {
-                        // all iso control characters are in U+00xx
-                        toAppendTo.setCharAt(i, '\\');
-                        toAppendTo.insert(i + 1, "u0000");
-                        toAppendTo.setCharAt(i + 4, Chars.getUpperCaseHex((c & 0xF0) >> 4));
-                        toAppendTo.setCharAt(i + 5, Chars.getUpperCaseHex(c & 0xF));
-                    } else if (c == '"' || c == '\\') {
-                        // only " and \ need to be escaped; other escapes are optional
-                        toAppendTo.insert(i, '\\');
-                    }
-                }
-            }
-        },
-
-        CRLF {
-            @Override
-            void escape(final StringBuilder toAppendTo, final int start) {
-                for (int i = toAppendTo.length() - 1; i >= start; i--) { // backwards: length may change
-                    final char c = toAppendTo.charAt(i);
-                    switch (c) {
-                        case '\r':
-                            toAppendTo.setCharAt(i, '\\');
-                            toAppendTo.insert(i + 1, 'r');
-                            break;
-                        case '\n':
-                            toAppendTo.setCharAt(i, '\\');
-                            toAppendTo.insert(i + 1, 'n');
-                            break;
-                    }
-                }
-            }
-        };
-
-        /**
-         * Escapes text using a standardized format from a given starting point to the end of the string.
-         *
-         * @param toAppendTo string buffer to escape
-         * @param start      where to start escaping from
-         */
-        abstract void escape(final StringBuilder toAppendTo, final int start);
-    }
```

Comment:
Alternatively I can pull the json implementation into a shared utility class elsewhere.